pipeline/README.md

```diff
@@ -0,0 +1,152 @@
+
+    oc create namespace uncontained-ci-cd
+    oc create namespace uncontained-development
+    oc create namespace uncontained-production
+
+## Configuration
+
+To configure your own application and send it through the pipeline you must update the `values.yaml` file to reflect a repository to which you have access. Please use a repository name with a lowercase format.
+
+It will be in this repository that the webhook will be automatically created and will trigger, on each new commit, a new PipelineRun on Tekton. You can create a new token on Github by [clicking here](https://github.com/settings/tokens), make sure to tick the `admin:repo_hook`, `repo` and `user` permissions.
+
+### Secrets 
+
+The Github token will only be shown once, be sure to save it somewhere safe. 
+Then copy the token and replace it in the command below.
+
+    oc create secret generic github-webhook-secret --from-literal=token=XXXXXXXXXXXXXXXX -n uncontained-ci-cd
+
+In order to avoid using passwords to publish new content on Github, we will use a keypair. 
+[Here's](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) how to create one if you don't already have one available on your machine.
+
+Assuming your private key was created in the default location `$HOME/.ssh/id_rsa`, you can run the command below to create a generic secret.
+
+    oc create secret generic uncontained-github-deploy-secret \
+        --from-file=ssh-privatekey=$HOME/.ssh/id_rsa \
+        --namespace uncontained-ci-cd
+
+### Applying
+
+Great. Now that the secrets have been set up correctly we can install the package. For this, we will use helm, as below.
+
+    helm template -f pipeline/values.yaml pipeline | oc apply -f-
+
+How to verify that the installation was successful:
+
+- See if you github webhook integration was created. Go to your github repository, click on Settings, and then on Webhooks.
+
+You can check the webhook creation logs by searching for `webhook` in your pods in the `uncontained-ci-cd` namespace. If something has failed, you can delete TaskRun `create-uncontained-github-webhook`, update your webhook data from `values.yaml` and run the **Applying** step again.
+
+### Policies
+
+Before publishing a new commit, we will need to define some permissions for the pipeline ServiceAccount. This will allow the pipeline to move images from one environment to another.
+
+```
+oc policy add-role-to-user \
+    system:image-puller system:serviceaccount:uncontained-ci-cd:uncontained-tekton-triggers-sa \
+    --namespace=uncontained-development
+```
+
+```
+oc policy add-role-to-user \
+    system:image-puller system:serviceaccount:uncontained-ci-cd:uncontained-tekton-triggers-sa \
+    --namespace=uncontained-production
+```
+
+And also for triggers ServiceAccount:
+
+    oc adm policy add-scc-to-user privileged system:serviceaccount:uncontained-ci-cd:uncontained-tekton-triggers-sa -n uncontained-ci-cd
```

Comment:
why does a pipeline require the privileged scc?

Reply:
It does not. It was only in the documentation and has been removed